onnxruntime/core/providers/migraphx/migraphx_execution_provider.cc

```diff
@@ -333,7 +348,7 @@ static bool can_eval_shape_general(const GraphViewer& graph, const Node* node, c
 
     // find the node corresponding to the name
     auto nit = std::find_if(in_nodes.begin(), in_nodes.end(), [&](auto n) {
-      return input_name.find(n->Name()) != std::string::npos;
+      return is_input_node(n, input_name);
```

Comment:
Is there a test for this function can_eval_shape_general?

Reply:
For now, we do not have a test for that.

Reply:
Shall we add a test to exercise the bug fix part? This part of the code seems error-prone, having some tests would be helpful.